### PR TITLE
[refactor] Refactor benchmark.dart

### PR DESCRIPTION
**Pull request by d-uzlov**
Part of https://github.com/mlcommons/mobile_app_open/issues/172

Another attempt to simplify `benchmark.dart`.

### CODE REVIEW

**Comment by d-uzlov** [comment]
I really don't like `Benchmark` and `BenchmarkJob` classes and methods that use them [of which there are a lot of].
I tried to refactor them but it seems that requires too many too complex changes.
In this PR I tried to simplify work with these classes and related methods a bit.
There is still work to be done to make it easy to read and modify.

File: flutter/lib/benchmark/benchmark.dart [hunk 1]
```diff
@@ -531,27 +461,32 @@ enum DatasetMode { lite, full, test }
 
 class BenchmarkJob {
   final Benchmark benchmark;
-  final pb.DatasetConfig dataset;
+  late final pb.DatasetConfig dataset;
   final bool accuracy;
-  final bool fast;
-  final DatasetMode _datasetMode;
+  late final bool fast;
```
Comment (anhappdev) [inline flutter/lib/benchmark/benchmark.dart:466]: `fast` can be private. 
Rename to `_fastMode` would be more meaningful. Please also rename `accuracy` to `accuracyMode`

File: flutter/lib/app_constants.dart [hunk 2]
```diff
@@ -3,6 +3,8 @@ import 'package:flutter/material.dart';
 const OFFICIAL_BUILD =
     bool.fromEnvironment('official_build', defaultValue: false);
 
+const FAST_MODE = bool.fromEnvironment('fast-mode', defaultValue: false);
```
Comment (anhappdev) [inline flutter/lib/app_constants.dart:6]: Please rename `fast-mode` to `fast_mode`, or rename `official_build` to `official-build`, so we are consistent in using either underscore or hyphen.

**Review by anhappdev — CHANGES_REQUESTED** [review]

File: flutter/lib/backend/list.dart [hunk 3]
```diff
@@ -12,3 +15,33 @@ List<String> getBackendsList() {
...
   }
 }
+
+class BackendInfo {
+  final pb.BackendSetting settings;
+  final String libPath;
+
+  BackendInfo._(this.settings, this.libPath);
+
+  static BackendInfo finMatching() {
```
[the remark this answers is not in the document]
Comment (d-uzlov) [reply flutter/lib/backend/list.dart]: Fixed

File: flutter/lib/benchmark/benchmark.dart [hunk 4]
```diff
@@ -211,75 +184,48 @@ class BenchmarkState extends ChangeNotifier {
...
-        errors.add(testDataPath);
-      }
-      final groundtruthDataPath = job.dataset.groundtruthSrc;
-      if (isInternetResource(groundtruthDataPath)) {
-        errors.add(groundtruthDataPath);
-      }
-    }
-
+    final resources = listSelectedResources();
+    final errors = filterInternetResources(resources);
```
[the remark this answers is not in the document]
Comment (d-uzlov) [reply flutter/lib/benchmark/benchmark.dart]: Done

File: flutter/lib/benchmark/benchmark.dart (hunk 1 above)
Comment (d-uzlov) [reply flutter/lib/benchmark/benchmark.dart]: Done

File: flutter/lib/app_constants.dart (hunk 2 above)
Comment (d-uzlov) [reply flutter/lib/app_constants.dart:6]: Done. I renamed `official_build` to `official-build`

**Review by anhappdev — APPROVED** [review]

**Pull request merged**